### ISSUE TITLE
S2228, S2278, S2070 throws null ref exception on invalid code

### ISSUE THREAD

**Issue by duncanpMS** [SonarSource/sonarlint-vs#451]
**Repro:** enter the following (invalid) code:
    `public decimal fieldD6 = fieldD5;`

**Expected:** no error

**Actual:** the ConsoleLogging analyser throws a null ref exception:

*"Message	AD0001	The User Diagnostic Analyzer 'SonarLint.Rules.ConsoleLogging' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.	ConsoleApplication1		1"*

**Comment by tvsonar** [comment]
@duncanpMS I couldn't reproduce the issue this way. But the sample you sent yesterday produces the expected error.
```
public decimal fieldD6 = int()
```

**Comment by duncanpMS** [comment]
@tvsonar Oops - I copied the wrong bit of invalid code when creating the issue. My mistake.
